Sources/CIImage+TestPattern.swift

```diff
@@ -0,0 +1,268 @@
+
+        // Start with an empty image and composite the pattern components on top.
+        var pattern = CIImage.empty()
+
+        // Add wide gamut color swatches in a grid in the lower part of the pattern.
+        for (column, rowColors) in self.swatchColors.enumerated() {
+            // Note: We reverse here so the first color specified in the array is displayed on the top.
+            for (row, color) in rowColors.reversed().enumerated() {
+                var swatch = CIImage.colorSwatch(for: color)
+                swatch = swatch.moved(to: CGPoint(x: CGFloat(column) * (swatch.extent.width + self.margin), y: CGFloat(row) * (swatch.extent.height + self.margin)))
+                pattern = swatch.composited(over: pattern)
+            }
+        }
+
+        // Add the brightness scale above the color swatches.
+        var brightnessScale = CIImage.brightnessScale(levels: self.brightnessScaleLevels)
+        brightnessScale = brightnessScale.moved(to: CGPoint(x: 0, y: pattern.extent.maxY + self.margin))
+        pattern = brightnessScale.composited(over: pattern)
+
+        // Add a nice title to the pattern.
+        var title = self.titleLabel()
+        title = title.moved(to: CGPoint(x: 0, y: pattern.extent.maxY + self.margin))
+        pattern = title.composited(over: pattern)
+
+        // Add the given text label at the bottom left corner of the image, if any.
+        if let labelText = label, !labelText.isEmpty {
+            var label = CIImage.text(labelText, fontName: self.labelFont, fontSize: 30, color: .white)!
+            label = label.moved(to: CGPoint(x: 0, y: -label.extent.height - self.margin))
+            pattern = label.composited(over: pattern)
+        }
+
+        // Put everything on a black background.
+        let background = CIImage(color: .black).cropped(to: pattern.extent.insetBy(dx: -self.margin, dy: -self.margin))
+        return pattern.composited(over: background).moved(to: .zero)
+    }
+
+
+    // MARK: - Internal
+
+    /// The size of the single color/brightness tiles.
+    private static var tileSize: CGSize { CGSize(width: 160, height: 160) }
+    /// The corner radius that is applied to color swatches and brightness scale.
+    private static var tileCornerRadius: Double { 10 }
+    /// The margin between elements in the pattern.
+    private static var margin: CGFloat { 50 }
+    /// The font to use for labels in the pattern.
+    private static var labelFont: String { ".AppleSystemUIFontCompactRounded-Semibold" }
+    /// Colors for which swatches should be rendered, ordered by columns (outer) and rows (inner arrays).
+    private static var swatchColors: [[CIColor]] { [[.red, .green, .blue], [.cyan, .magenta, .yellow]] }
+    /// The different color spaces in which colors should be rendered in the color swatches.
+    private static var swatchColorSpaces: [(label: String, colorSpace: CGColorSpace)] = [
```

Comment:
Would it make sense to have some kind of `Swatch` type with label, color space and color?

Reply:
This dict here is only needed to assign labels to the different color spaces. What would the `Swatch` type replace then?